Centralize how gvproxy PID and Log paths are calculated

lstocchi opened the pull request:
The gvproxy pid and log files were consistently generated using the same code in different places.
This patch centralizes them into two func.

related to https://github.com/crc-org/macadam/issues/225

cfergeau commented on pkg/machine/gvproxy.go:
```diff
@@ -29,3 +30,11 @@ func CleanupGVProxy(f define.VMFile) error {
+}
+
+func GetGVProxyLogFile(mc *vmconfigs.MachineConfig, provider vmconfigs.VMProvider, dirs *define.MachineDirs) (*define.VMFile, error) {
+	return dirs.RuntimeDir.AppendToNewVMFile(fmt.Sprintf("gvproxy-%s-%s.log", mc.Name, provider.VMType()), nil)
```
Wondering if `RuntimeDir` should be modified to contain the vm type, similarly to what is done for `DataDir`, eg `/run/user/1000/macadam/qemu/` on linux instead of `/run/user/1000/macadam/`?

cfergeau replied:
(just a comment, no change required) The `dirs` args could be removed and replaced by a call to `mc.RuntimeDir()`, but since most of the callers already have a `dirs`  variable, it’s probably better to just reuse it as you did.

cfergeau replied:
And I just got confused by https://github.com/containers/podman/blob/0615fdecfcfcfeb931a55458a818dfca42f6b365/pkg/machine/vmconfigs/config.go#L18 which has pid file/log file members which are persisted in the machine json description, but appear to be unused.

One small impact of this change is that starting a macadam vm, upgrading macadam and then trying to stop it won’t be 100% smooth as macadam won’t be able to find the gvproxy pid file (but I’m not sure it is that important).

cfergeau commented on pkg/machine/vmconfigs/machine.go:
```diff
@@ -259,6 +259,21 @@ func (mc *MachineConfig) RuntimeDir() (*define.VMFile, error) {
+	default:
+		return define.QemuVirt
+	}
+}
```
Might be better to compute the runtime file suffix when creating the MachineConfig instead of guessing it. We have a VMType at the time it’s created.
My comment about using `run/user/1000/macadam/qemu/` instead of `/run/user/1000/macadam/` (ie adding a `$vmtype’ subdirectory) may be more relevant in this commit than in the previous.

cfergeau reviewed (commented):
Some discussion points, though imo the first commit could go as is for now to fix multiple VM issues, it can be refined in follow-ups.

lstocchi committed:
add helper funcs to generate gvproxy pid and log filenames

The gvproxy pid and log files were consistently generated using the same code in different places.
This patch centralizes them into two func.

Signed-off-by: lstocchi <lstocchi@redhat.com>

lstocchi commented:
@cfergeau your feedback made me think .... technically we only need to update the runtime dir and we can already do this from macadam 

https://github.com/crc-org/macadam/blob/5ae52b4a3b6375c5f4b60890986f2ca86a7f8a08/pkg/env/env.go#L37

so this patch is not actually needed. 

I just kept the code to centralize how the file are created to make it cleaner.

I did not add the provider folder creation here bc as you said it could cause issues if someone has a started machine. This way it would be easier for the podman team to merge the code as it is.
For macadam i don't think we are going to cause much trouble.

So in conclusion, i'm going to add a PR in macadam to have something like 
```
err = os.Setenv("PODMAN_RUNTIME_DIR", filepath.Join("macadam", provider.VMType().String()))
```
and we should be good
